app/assets/javascripts/watchlist.js

```diff
@@ -0,0 +1,420 @@
+    });
+
+    var instance_ids = [];
+    selected_user_ids.forEach(user_id => {
+      instance_ids = instance_ids.concat(new_instances[user_id]["instance_ids"]);
+    });
+  
+    if (instance_ids.length > 0) {
+      window.open(`mailto: ${emails}`, "_blank");
+      update_watchlist(method, instance_ids);
+    }
+  });
+
+  $('#resolve_button').click(function(){
+    method = "resolve";
+    var instances = get_active_instances(new_instances, contacted_instances);
+
+    var instance_ids = [];
+    selected_user_ids.forEach(user_id => {
+      instance_ids = instance_ids.concat(instances[user_id]["instance_ids"]);
+    });
+
+    if (instance_ids.length > 0) {
+      update_watchlist(method, instance_ids);
+    }
+  });
+}
+
+// Uncheck all checkboxes when moving to another tab
+$('.ui.vertical.fluid.tabular.menu .item').on('click', function() {
+  $('.ui.tab.segments.active .ui.checkbox').checkbox('uncheck');
+  $('.ui.checkbox.select_all').checkbox('uncheck');
+  $('.ui.vertical.fluid.tabular.menu .item').removeClass('active');
+  $(this).addClass('active');
+
+  switch ($(this).attr("data-tab")) {
+    case "new_tab":
+      $("#contact_button").removeClass("disabled");
+      $("#resolve_button").removeClass("disabled");
+      break;
+    case "contacted_tab":
+      $("#contact_button").addClass("disabled");
+      $("#resolve_button").removeClass("disabled");
+      break;
+    case "resolved_tab":
+      $("#contact_button").addClass("disabled");
+      $("#resolve_button").addClass("disabled");
+      break;
+    case "archived_tab":
+      $("#contact_button").addClass("disabled");
+      $("#resolve_button").addClass("disabled");
+      break;
+    default:
+      console.log(`${$(this).attr("data-tab")} is not a valid tab`);
+      return;
+  }
+});
+
+function get_active_instances(new_instances, contacted_instances) {
+  var tab = $(".ui.tab.segments.active").attr('id');
+  
+  switch(tab){
+    case "new_tab":
+      return new_instances;
+    case "contacted_tab":
+      return contacted_instances;
+    default:
+      console.log(`${tab} is not a valid tab for this action`)
+      return;
+  }
+}
+
+function update_watchlist(method, ids){
+	let students_selected = {};
+	students_selected['method'] = method;
+	students_selected['ids'] = ids;
+
+	$.ajax({
+		url:watchlist_endpoints['update'],
+		dataType: "json",
+		contentType:'application/json',
+		data: JSON.stringify(students_selected),
+		type: "POST",
+		success:function(data){
+			refresh_watchlist();
+		},
+		error:function(result, type){
+			render_banner({
+				type:"negative",
+				header:"Currently unable to " + method + " students",
+				message: "Do try again later",
+				timeout: -1
+			});
+		},
+		complete:function(){
+		}
+	});
+}
+
+// instructor clicks on 'refresh' button
+$('#refresh_btn').click(function(){
+	refresh_watchlist();
+});
+
+function refresh_watchlist(){
+	$.getJSON(watchlist_endpoints['get'],function(data, status){
```

Comment:
Should this be watchlist_endpoints['refresh']? I was trying to click the refresh button and it only seems to re-obtain the current watchlist instances.

Reply:
Thanks for catching that!